head/firmware/can.c

```diff
@@ -6,22 +6,27 @@
  * @param handle Pointer to an FDCAN handle
  * @return HAL_OK on success
  */
-HAL_StatusTypeDef MX_FDCAN1_Init(FDCAN_HandleTypeDef* handle) {
+HAL_StatusTypeDef MX_FDCAN1_Init(
+    FDCAN_HandleTypeDef* handle,
+    uint8_t clock_divider,
+    uint8_t segment_1_tqs,
+    uint8_t segment_2_tqs,
+    uint8_t max_sync_jump_width) {
     handle->Instance = FDCAN1;
     handle->Init.ClockDivider = FDCAN_CLOCK_DIV1;
     handle->Init.FrameFormat = FDCAN_FRAME_FD_NO_BRS;
     handle->Init.Mode = FDCAN_MODE_NORMAL;
     handle->Init.AutoRetransmission = ENABLE;
     handle->Init.TransmitPause = DISABLE;
     handle->Init.ProtocolException = DISABLE;
-    handle->Init.NominalPrescaler = 20;
-    handle->Init.NominalSyncJumpWidth = 2;
-    handle->Init.NominalTimeSeg1 = 14;
-    handle->Init.NominalTimeSeg2 = 2;
-    handle->Init.DataPrescaler = 20;
-    handle->Init.DataSyncJumpWidth = 1;
-    handle->Init.DataTimeSeg1 = 14;
-    handle->Init.DataTimeSeg2 = 1;
+    handle->Init.NominalPrescaler = clock_divider;
+    handle->Init.NominalSyncJumpWidth = max_sync_jump_width;
```

Comment:
Curious if using the max value here is correct?

Reply:
it's just a naming thing. The `Nominal` in the fdcan init structure is contrasted to `Data`, not `max`  - it's for the flexible datarate thing you can do, where you start off at the nominal timings and once you've transferred enough of the arbitration id and header you switch to the Data timings.